src/scripts/background/connection/callbacks/ProtocolCallbacks.ts

```diff
@@ -17,10 +20,45 @@ async function addWebpage([url]: [string]): Promise<void> {
   await getMintingRegistrar().maybeTryRegister();
 }
 
+async function isWhitelisted(): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    chrome.tabs.query(
+      { currentWindow: true, active: true },
+      function (tabArray) {
+        let currentUrl = new URL(tabArray[0].url as string);
+        if (FACT_ACCESS_WHITELIST.includes(currentUrl.host)) {
+          resolve(true);
+        } else {
+          reject("Access denied.");
+        }
+      },
+    );
+  });
+}
```

Comment:
I would have `isWhitelisted` only fail if some error happens accessing the chrome.tab. You can change the name to `requireWhitelisted` and either fail or resolve with nothing.